### PR TITLE
Allow only registered users to post a job

### PR DESCRIPTION
**Pull request by RanaWaqar7550**
### Description

[LP-2507](https://philanthropyu.atlassian.net/browse/LP-2507)

[philu-edx-theme PR](https://github.com/philanthropy-u/philu-edx-theme/pull/1436)

### CODE REVIEW

File: openedx/features/job_board/views.py [hunk 1]
```diff
@@ -114,9 +117,20 @@ class JobDetailView(DetailView):
...
+    Create job view and submit job posting data
+    """
     form_class = JobCreationForm
     template_name = 'features/job_board/create_job_form.html'
 
+    def get_initial(self, *args, **kwargs):  # pylint: disable=unused-argument, arguments-differ
+        """Pre-fill form with initial data"""
+        initial = super(JobCreateView, self).get_initial(**kwargs)
+        user = self.request.user
+        initial['user'] = user.profile.user
```
Comment (harris-irfan) [inline openedx/features/job_board/views.py:132]: why are we assigning `user.profile.user` instead of just `user`

**Review by aliassad — APPROVED** [review]
LGTM 👍

File: openedx/features/job_board/views.py (hunk 1 above)
Comment (RanaWaqar7550) [reply openedx/features/job_board/views.py:132]: I have to save user extended profile data, just like we are doing for `idea` feature.
https://github.com/philanthropy-u/edx-platform/blob/c531c6565a92a901ce82dad5d8c9ff329fab9fab/openedx/features/idea/views.py#L70

File: openedx/features/job_board/views.py [hunk 2]
```diff
@@ -114,9 +117,20 @@ class JobDetailView(DetailView):
...
+    Create job view and submit job posting data
+    """
     form_class = JobCreationForm
     template_name = 'features/job_board/create_job_form.html'
 
+    def get_initial(self, *args, **kwargs):  # pylint: disable=unused-argument, arguments-differ
+        """Pre-fill form with initial data"""
+        initial = super(JobCreateView, self).get_initial(**kwargs)
+        user = self.request.user
+        initial['user'] = user
```
Comment (harris-irfan) [inline openedx/features/job_board/views.py:132]: in this case now, we can also directly `initial['user'] = self.request.user`

Just a casual suggestion, feel free to ignore it.

**Review by harris-irfan — APPROVED** [review]

**Pull request merged**